### ISSUE TITLE
fileInfo not working (stays locked to record used first) ??

### ISSUE THREAD

**Issue by wesyah234** [tomitrescak/meteor-uploads#133]
I hope I can explain this clearly.

I have an app where I can view multiple records and choose one of them to open up.  I use Iron router and when opening up a record, it's a route like /viewRecord/1234 if I'm opening record 1234.  I have the ability to open a second record, say /viewRecord/5678, and I can flip between them on the application, changing the route and getting the other record displayed.  On each viewRecord screen, I have the uploader control, allowing one to upload files that will be attached to the respective record.  To attach them properly, I have a helper:

```
fileUploadInfo: function () {
    // this is data that will be passed to the server with the upload
    console.log("getting file upload info this is: ", this);
    var toReturn = {chartId: this._id, userName: Meteor.user().username};
    console.log("returning ", toReturn);
    return toReturn;
  }
```
As I flip between the 2 open records, I see the console logging showing that it's returning the object with the chartId and userName set properly for the differing chartIds (1234 vs 5678).

I am saving the file metadata to a collection where I will have the chartId saved as one of the fields, so I know which files are associated with which records.

I have found during testing that if I upload a file to one chart, it works fine, but if I then switch to a different chart record, and upload a file, it still thinks it's on the original chart and the fileInfo received by the "finished" callback is wrong.

In other words, I open chart 1234, upload a file, and it knows it's working with chart 1234 and it saves the metadata correctly.  But then if I open chart 5678, and upload a file, it still thinks it's working with chart 1234 and saves the metadata incorrectly, effectively attaching the second file to the first chart (1234).

Also, perhaps related is the fact that when I switch to the second record, the uploader control still shows the last filename uploaded... so, I think the control doesn't know there should be effectively 2 "instances" of itself...?? does this make sense?  Any ideas?  Thank you!

**Comment by wesyah234** [comment]
Update: I've realized that this only happens when I "stay" on the same route, for example, if I'm viewing route /viewRecord/1234 and then upload to that chart, then view route /viewRecord/5678, the upload control is "stuck" at record 1234, however, if after viewing record 1234, I go back to the main route like /viewAllRecords, then go to record 5678, the upload control is reinitialized (ie. it does not show the left over filename from the last upload), and then it also does not upload to the wrong record.

**Comment by tomitrescak** [comment]
Hello, this seems like a standard behaviour.The upload component is initialised in `rendered` callback. If rendered callback does not happen when you are changing routes (which you are not, you are only changing context), the component will hold the last used values. I'm not sure if I'm explaining myself.

The easiest fix for you is to register the upload component in the global namespace and call [Uploader.reset](https://github.com/tomitrescak/meteor-tomi-upload-jquery/blob/master/uploader.js#L140-L144) when you are changing routes. 

I'll close this for now, please reopen if it did not help. 
Sorry for the late reply.

**Comment by wesyah234** [comment]
Can you explain a bit further? Where exactly would i call the reset function?

Is there any way you can fix this ?  I would think this would be a common usecase to move between different records and want the upload component to know about the change of records and not be "stuck" on the previous record...
